### PR TITLE
Blog on `Journey With Go - A Blog Series`

### PR DESCRIPTION
**Pull request by grgprarup**
This PR adds the first episode of `Journey With Go - A Blog Series`.

### CODE REVIEW

File: src/assets/TestAutomationWithGo-ABlogSeries/goBasics.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+authorAvatar: https://www.jankaritech.com/images/2021/12/29/p1110938-crop-colour-800.jpg
+authorLink: https://github.com/grgprarup
+createdAt: Dec 10, 2022
+tags: golang, gopher, go, test automation, testing, bdd, godog
+banner: https://camo.githubusercontent.com/2b507540e2681c1a25698f246b9dca69c30548ed66a7323075b0224cbb1bf058/68747470733a2f2f676f6c616e672e6f72672f646f632f676f706865722f6669766579656172732e6a7067
+seriesTitle: Journey With Go - A Blog Series
+episode: 1
+---
+
+**गो** **"Journey With Go - A Blog Series"** about the basics of Go `Go Basic`, advanced concepts in Go `Go Beyond Basics`, testing with go `Godog as Test Framework` and many more. This is the first blog of the series. In this blog, we will learn about the installation of Go, initializing a Go project. We will also learn about the basic data types, variables, constants in Go. So, let's get started.
```
Comment (SwikritiT) [inline src/assets/TestAutomationWithGo-ABlogSeries/goBasics.md:13]:
```suggestion
**गो** **"Journey With Go - A Blog Series"** about the basics of Go `Go Basic`, advanced concepts in Go `Go Beyond Basics`, testing with go `Godog as Test Framework` and many more. This is the first blog of the series. In this blog, we will learn about the installation of Go, and initializing a Go project. We will also learn about the basic data types, variables, and constants in Go. So, let's get started.
```

File: src/assets/TestAutomationWithGo-ABlogSeries/goBasics.md [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+
+## Introduction
+- Designed and Developed at Google by Robert Griesemer, Rob Pike, and Ken Thompson
+- Go is an open source programming language supported by Google
+- Go is a statically typed i.e. you need to declare type of variable before using it
+- Go is compiled programming language i.e. you need to compile your code before running it
```
Comment (SwikritiT) [inline src/assets/TestAutomationWithGo-ABlogSeries/goBasics.md:19]:
```suggestion
- Designed and Developed at Google by Robert Griesemer, Rob Pike, and Ken Thompson
- Go is an open-source programming language supported by Google
- Go is statically typed i.e. you need to declare the type of a variable before using it
- Go is compiled programming language i.e. you need to compile your code before running it
```

File: src/assets/TestAutomationWithGo-ABlogSeries/goBasics.md (hunk 1 above)
Comment (kiranparajuli589) [inline src/assets/TestAutomationWithGo-ABlogSeries/goBasics.md:13]:
```suggestion
**गो** **"Journey With Go - A Blog Series"** about the basics of Go `Go Basic`, advanced concepts in Go `Go Beyond Basics`, testing with go `Godog as Test Framework` and many more. This is the first blog in the series. In this blog, we will learn about installing Go and initializing a Go project. We will also learn about basic data types, variables, and constants in Go. So, let's get started.
```

File: src/assets/TestAutomationWithGo-ABlogSeries/goBasics.md [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+### Verifying Installation
+To verify the installation, you can run the following command to check the version of Go.
+```go
+go version
+```
+![image](/src/assets/TestAutomationWithGo-ABlogSeries/images/go-version.png)
+You can see the version of Go as shown in the image. Here, the version of Go is 1.19.4.
+
+### Initializing a Go Project
+Go uses the `go mod` command to initialize a Go project. The `go mod` command creates a go.mod file in the current directory. The go.mod file contains the information about the project and its dependencies.
```
Comment (SwikritiT) [inline src/assets/TestAutomationWithGo-ABlogSeries/goBasics.md:72]:
```suggestion
Go uses the `go mod` command to initialize a Go project. The `go mod` command creates a `go.mod` file in the current directory. The `go.mod` file contains information about the project and its dependencies.
```

File: src/assets/TestAutomationWithGo-ABlogSeries/goBasics.md [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+- Declare a variable and assign a value to it later
+
+##### Declaring and Assigning a Value to a Variable
+
+**1. Declaring single variable**
+The following code declares a variable named `number` and assigns a value to it.
+```go
+var number int = 10
+```
+The similar code can be written as follows.
```
Comment (SwikritiT) [inline src/assets/TestAutomationWithGo-ABlogSeries/goBasics.md:158]:
```suggestion
A similar code can be written as follows:
```

File: src/assets/TestAutomationWithGo-ABlogSeries/goBasics.md [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+```go
+var number = 10
+```
+In such type of declaration Go automatically infers the type of the variable from the value assigned to it.
+
+The variable can be declared without the `var` keyword as follows.
```
Comment (SwikritiT) [inline src/assets/TestAutomationWithGo-ABlogSeries/goBasics.md:164]:
```suggestion
The variable can be declared without the `var` keyword as follows:
```

File: src/assets/TestAutomationWithGo-ABlogSeries/goBasics.md [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+```go
+number := 10
+```
+**2. Declaring multiple variables**
+The following code declares two variables named `firstNumber` and `secondNumber`
+```go
+var firstNumber, secondNumber int = 10, 20
+```
+The similar code can be written as follows.
```
Comment (SwikritiT) [inline src/assets/TestAutomationWithGo-ABlogSeries/goBasics.md:173]:
```suggestion
A similar code can be written as follows:
```

File: src/assets/TestAutomationWithGo-ABlogSeries/goBasics.md [hunk 7]
```diff
@@ -0,0 +1,228 @@
...
+```go
+var firstNumber, secondNumber = 10, 20
+```
+The variable can be declared without the `var` keyword as follows.
```
Comment (SwikritiT) [inline src/assets/TestAutomationWithGo-ABlogSeries/goBasics.md:177]:
```suggestion
The variable can be declared without the `var` keyword as follows:
```

File: src/assets/TestAutomationWithGo-ABlogSeries/goBasics.md [hunk 8]
```diff
@@ -0,0 +1,228 @@
...
+
+##### Declaring and Assigning a Value to a Variable Later
+
+The following code declares a variable named name and assigns a value to it later.
+```go
+var name string
+
+name = "Prarup Gurung"
+```
+But we cannot declare a variable with the `:=` syntax and assign a value to it later. Also, we cannot declare a variable without keyword `var` and assign a value to it later.
```
Comment (SwikritiT) [inline src/assets/TestAutomationWithGo-ABlogSeries/goBasics.md:190]:
```suggestion
But we cannot declare a variable with the `:=` syntax and assign a value to it later. Also, we cannot declare a variable without the keyword `var` and assign a value to it later.
```

File: src/assets/TestAutomationWithGo-ABlogSeries/goBasics.md [hunk 9]
```diff
@@ -0,0 +1,228 @@
...
+
+#### Declaring a Constant
+
+The following code declares a constant named name and assigns a value to it.
+```go
+const name string = "Prarup Gurung"
+```
+#### Declaring Multiple Constants
+
+You can declare multiple constants in a single line. The following code declares two constants named name, address.
```
Comment (SwikritiT) [inline src/assets/TestAutomationWithGo-ABlogSeries/goBasics.md:204]:
```suggestion
You can declare multiple constants in a single line. The following code declares two constants named name, and address.
```

File: src/assets/TestAutomationWithGo-ABlogSeries/goBasics.md [hunk 10]
```diff
@@ -0,0 +1,228 @@
+---
+title: Go Basics - The Starting Point
+authorName: Prarup Gurung(प्रारुप गुरुङ्ग)
```
Comment (saw-jan) [inline src/assets/TestAutomationWithGo-ABlogSeries/goBasics.md:3]: :+1:

File: src/assets/TestAutomationWithGo-ABlogSeries/goBasics.md [hunk 11]
```diff
@@ -0,0 +1,228 @@
...
+banner: https://camo.githubusercontent.com/2b507540e2681c1a25698f246b9dca69c30548ed66a7323075b0224cbb1bf058/68747470733a2f2f676f6c616e672e6f72672f646f632f676f706865722f6669766579656172732e6a7067
+seriesTitle: Journey With Go - A Blog Series
+episode: 1
+---
+
+**गो** **"Journey With Go - A Blog Series"** about the basics of Go `Go Basic`, advanced concepts in Go `Go Beyond Basics`, testing with go `Godog as Test Framework` and many more. This is the first blog in the series. In this blog, we will learn about installing Go, and initializing a Go project. We will also learn about the basic data types, variables, and constants in Go. So, let's get started.
+
+## Introduction
+- Designed and Developed at Google by Robert Griesemer, Rob Pike, and Ken Thompson
+- Go is an open-source programming language supported by Google
```
Comment (saw-jan) [inline src/assets/TestAutomationWithGo-ABlogSeries/goBasics.md:17]: sounded like only supported by the google platforms
```suggestion
- Go is an open-source programming language backed by Google
```

File: src/assets/TestAutomationWithGo-ABlogSeries/goBasics.md [hunk 12]
```diff
@@ -0,0 +1,228 @@
...
+- Go is compiled programming language i.e. you need to compile your code before running it
+
+## Why Go?
+
+- Free and open source
+- Automatic Garbage Collection
+- Richer built-in types
+- Robust standard library
+- Built-in support for concurrency
+- Support function multiple return values
```
Comment (SagarGi) [inline src/assets/TestAutomationWithGo-ABlogSeries/goBasics.md:28]:
```suggestion
- Support function with multiple return values
```

File: src/assets/TestAutomationWithGo-ABlogSeries/goBasics.md [hunk 13]
```diff
@@ -0,0 +1,228 @@
...
+- Beautiful error handling
```
Comment (SagarGi) [inline src/assets/TestAutomationWithGo-ABlogSeries/goBasics.md:29]:
```suggestion
- Better error handling
```
or other words can be given?

File: src/assets/TestAutomationWithGo-ABlogSeries/goBasics.md [hunk 14]
```diff
@@ -0,0 +1,228 @@
...
+- CSP-style concurrency
+
+## Installation
+
+**Go** is available for any OS platform i.e. Windows, Linux, and Mac OS X. The installation is very simple. You can download the latest version of Go from [here](https://go.dev/doc/install). The latest version of Go at the time of writing this article is 1.19.4.
+
+Note: From here the overall setup will be done for Ubuntu 22.04. If you are using any other OS, please follow the [official documentation](https://go.dev/doc/install).
```
Comment (SagarGi) [inline src/assets/TestAutomationWithGo-ABlogSeries/goBasics.md:36]:
```suggestion
Note: From here the overall setup will be done in Ubuntu 22.04. If you are using any other OS, please follow the [official documentation](https://go.dev/doc/install).
```

File: src/assets/TestAutomationWithGo-ABlogSeries/goBasics.md (hunk 13 above)
Comment (grgprarup) [reply src/assets/TestAutomationWithGo-ABlogSeries/goBasics.md:29]: IMO this `Beautiful` word gives more weightage to the statement.

File: src/assets/TestAutomationWithGo-ABlogSeries/goBasics.md [hunk 15]
```diff
@@ -0,0 +1,228 @@
...
+You can see the version of Go as shown in the image. Here, the version of Go is 1.19.4.
+
+### Initializing a Go Project
+Go uses the `go mod` command to initialize a Go project. The `go mod` command creates a `go.mod` file in the current directory. The `go.mod` file contains information about the project and its dependencies.
+
+```go
+// go-basics is the name of the project/module
+go mod init go-basics
+```
+![image](/src/assets/TestAutomationWithGo-ABlogSeries/images/go-mod-init.png)
```
[the remark this answers is not in the document]
Comment (grgprarup) [reply src/assets/TestAutomationWithGo-ABlogSeries/goBasics.md:78]: Ohh!!! do we have such a feature? I will give it a try. Thanx @kiranparajuli589

**Review by saw-jan — COMMENTED** [review]
LGTM :+1:

**Review by SagarGi — APPROVED** [review]
LGTM :+1:

**Review by kiranparajuli589 — APPROVED** [review]

**Pull request merged**